components/style/properties/helpers/animated_properties.mako.rs

```diff
@@ -907,6 +908,32 @@ impl Animatable for i32 {
 }
 
 /// https://drafts.csswg.org/css-transitions/#animtype-number
+impl Animatable for NonNegativeNumber {
+    #[inline]
+    fn add_weighted(&self, other: &Self, self_portion: f64, other_portion: f64) -> Result<Self, ()> {
+        self.0.add_weighted(&other.0, self_portion, other_portion).map(NonNegativeNumber)
```

Comment:
So, we're not clamping here, and I think that's the intention, right? If so, some comments about it would be nice.